qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java

```diff
@@ -0,0 +1,204 @@
+    final var uri =
+        URI.create(
+            String.format(
+                "http://%s/%s*/_search",
+                elasticsearchContainer.getHttpHostAddress(), INDEX_PREFIX));
+    final var request = HttpRequest.newBuilder(uri).method("POST", BodyPublishers.noBody()).build();
+    final var response = httpClient.send(request, BodyHandlers.ofInputStream());
+    final var result = mapper.readValue(response.body(), EsSearchResponseDto.class);
+    if (result.esDocumentsWrapper == null) {
+      return Collections.emptyList();
+    }
+    return result.esDocumentsWrapper.documents.stream()
+        .map(esDocumentDto -> esDocumentDto.record)
+        .collect(Collectors.toList());
+  }
+
+  void startElasticsearch() {
+    final var version = RestClient.class.getPackage().getImplementationVersion();
+    elasticsearchContainer =
+        new ElasticsearchContainer(
+                DockerImageName.parse("docker.elastic.co/elasticsearch/elasticsearch")
+                    .withTag(version))
+            .withNetwork(network)
+            .withNetworkAliases(ELASTICSEARCH_HOST)
+            .withCreateContainerCmdModifier(
+                cmd ->
+                    Objects.requireNonNull(cmd.getHostConfig())
+                        .withMemory((long) (1024 * 1024 * 1024)));
+    elasticsearchContainer.start();
+  }
+
+  private void startZeebe() {
+    zeebeContainer =
+        new ZeebeContainer(ZeebeTestContainerDefaults.defaultTestImage())
+            .withNetwork(network)
+            .withEnv(
+                "ZEEBE_BROKER_EXPORTERS_ELASTICSEARCH_CLASSNAME",
+                "io.camunda.zeebe.exporter.ElasticsearchExporter")
+            .withEnv(
+                "ZEEBE_BROKER_EXPORTERS_ELASTICSEARCH_ARGS_URL",
+                String.format("http://%s:9200", ELASTICSEARCH_HOST))
+            .withEnv("ZEEBE_BROKER_EXPORTERS_ELASTICSEARCH_ARGS_BULK_DELAY", "1")
+            .withEnv("ZEEBE_BROKER_EXPORTERS_ELASTICSEARCH_ARGS_BULK_SIZE", "1")
+            .withEnv("ZEEBE_BROKER_EXPORTERS_ELASTICSEARCH_ARGS_INDEX_PREFIX", INDEX_PREFIX)
+            .withEnv("ZEEBE_CLOCK_CONTROLLED", "true");
+    zeebeContainer.start();
+    zeebeClient =
+        ZeebeClient.newClientBuilder()
+            .usePlaintext()
+            .gatewayAddress(zeebeContainer.getExternalGatewayAddress())
+            .build();
+    clockActuatorClient = new ClockActuatorClient(zeebeContainer.getExternalMonitoringAddress());
+  }
+
+  /**
+   * Wait until all records are exported. We assume that all records are exported if the number of
+   * records is does not change for 5 seconds. Initialize the counter to one so that we expect at
+   * least one record and don't succeed if nothing is exported.
+   */
+  private void waitForExportedRecords() {
+    final AtomicInteger previouslySeenRecords = new AtomicInteger(1);
+    Awaitility.await("Waiting for a stable number of exported records")
```

Comment:
:upside_down_face: I think there's a way to combine the `waitForExportedRecords` and the extra `searchExportedRecords` in the tests, since Awaitility offers a `<T> T Awaitility.await().until(Callable<T>, Predicate<T>)` variant which would return the latest results of the callable which you could return here. But maybe it's more complicated to use that way, and this is fine.

Reply:
Yeah I've tried that first but it got a bit messy